nucleus/src/main/java/za/co/absa/subatomic/application/team/TeamService.java

```diff
@@ -139,9 +143,33 @@ public void removeTeamMember(String teamId, String memberId,
                 actionedByEntity);
     }
 
-    public TeamEntity addSlackIdentity(String teamId, String teamChannel) {
-        return this.persistenceHandler.addSlackIdentity(teamId,
+    public TeamEntity addSlackIdentity(String teamId, String actionedByMemberId, String teamChannel) {
+        // actionedByMemberId (is the now the variable name standard) = createBy = requestedById
+        TeamEntity teamEntity = this.persistenceHandler.addSlackIdentity(teamId,
                 teamChannel);
+
+        TeamMemberEntity member = this.teamMemberPersistenceHandler
+                .findByTeamMemberId(actionedByMemberId);
+
+        TeamSlackIdentity teamSlackIdentity = new TeamSlackIdentity(
+                teamEntity.getSlackDetails().getTeamChannel());
+
+        TeamMemberSlackIdentity teamMemberSlackIdentity = new TeamMemberSlackIdentity(
+                member.getSlackDetails().getScreenName(),
+                member.getSlackDetails().getUserId());
+
+        AtomistTeam atomistTeam = new AtomistTeam (teamEntity.getTeamId(),
```

Comment:
Please use the AtomistTeamMapper.createAtomistTeam() function here instead.

Reply:
Also do not create the Atomist** stuff in the Service. It belongs in the Infrastructure layer -> i.e. the automationHandler class.